src/chatGPT.ts

```diff
@@ -1,152 +1,241 @@
-            return {
-                updatedContent: "",
-                createdKeys: [],
-            };
-        }
-
-        console.log(
-            "Full response text with delimiters from Azure OpenAI:",
-            responseText
-        ); // Log the full response text
-
-        // Extract the content between the delimiters
-        const componentsStart = responseText.indexOf("---T components---") + "---T components---".length;
-        const componentsEnd = responseText.indexOf("---KEYS---");
-        const fileContentWithComponents = responseText.substring(componentsStart, componentsEnd).trim();
-
-        console.log('Updated content with <T> components:', fileContentWithComponents); // Log the updated content
-
-        const responseParts = responseText?.split('---KEYS---'); // Split the response at a marker like `---KEYS---`
-        if (!responseParts) {
-            console.error("Failed to split the response text.");
-            return {
-                updatedContent: "",
-                createdKeys: [],
-            };
-        }
-
-        const keyListString = responseParts[1];  // The JSON-like string for the key list
-        if (!keyListString) {
-            console.error("Failed to access list of keys.");
-            return {
-                updatedContent: "",
-                createdKeys: [],
-            };
-        }
-        console.log("Key list string from ChatGPT:", keyListString); // Log the raw key list string
-
-
-        // Parse the key list from the response
-        let createdKeys: { keyName: string; description: string; translations: { en: string } }[] = [];
-        try {
-            createdKeys = extractCreatedKeys(keyListString); // Parse the list of keys
-            console.log("Parsed created keys:", createdKeys); // Log the parsed list of keys
-        } catch (error) {
-            console.error("Failed to parse the list of keys from ChatGPT response.");
-            createdKeys = [];
-        }
-
-        return {
-            updatedContent: fileContentWithComponents,
-            createdKeys,
-        };
+                            3. **Do not modify any content already wrapped in a <T> component**. Ensure that existing <T> components are accurately detected and skipped.
+                            4. **Do not modify strings inside attributes or event handlers that use localization functions like useTranslate("key name") or t()**. Leave them unchanged.
+                            5. **Only replace plain text inside DOM elements like <div>Some text</div> or <button>Some text</button> with <T> components**.
+                            6. Replace plain text strings with the <T> component, using key names with dashes (-).
+                            7. Replace plain text strings inside non-component contexts like attributes (e.g., placeholder, title) with useTranslate("key name") and not {useTranslate("key name")}.
```

Comment:
It doesn't work this way. Please read the docs. https://tolgee.io/js-sdk/integrations/react/translating